src/main/java/net/minecraftforge/event/TickEvent.java

```diff
@@ -66,6 +67,25 @@ public WorldTickEvent(LogicalSide side, Phase phase, Level world)
             this.world = world;
         }
     }
+
+    /**
+     * Called before and after a chunk is ticked for things like random ticks and weather. Useful for random-tick and weather based actions.
+     */
+    public static class EnvironmentTickEvent extends TickEvent
+    {
+        public final Level level;
+        public final LevelChunk chunk;
+        public final int randomTickSpeed;
```

Comment:
Please expose these as getters and mark these `private`. I've been meaning to refactor these tick events to encapsulate the fields, but I keep missing my windows to do them.